components/Graph.tsx

```diff
@@ -0,0 +1,157 @@
+import { FText } from '~/components/Text/FText';
+
+interface DataPoint {
+  value: number;
+  label: string;
+}
+
+const Graph = () => {
+  // Generate mock data only once
+  // const [allData] = useState<DataPoint[]>(
+  //   Array.from({ length: 365 }, (_, i) => ({
+  //     value: Math.floor(Math.random() * (2763 - 1000 + 1)) + 1000, // Random price
+  //     label: new Date(Date.now() - i * 24 * 60 * 60 * 1000).toISOString(), // ISO date
+  //   }))
+  // );
+
+  const [allData] = useState<DataPoint[]>([]);
+
+  const [selectedRange, setSelectedRange] = useState<string>('1month');
+  const [filteredData, setFilteredData] = useState<DataPoint[]>([]);
+  const [currentValue, setCurrentValue] = useState<number>(0);
+  const [currentDate, setCurrentDate] = useState<string>('');
+  const [containerWidth, setContainerWidth] = useState<number>(
+    Dimensions.get('window').width - 32 // Default width with padding
+  );
+
+  // Filter data based on selected range
+  useEffect(() => {
+    const now = new Date();
+    let range = 0;
+
+    switch (selectedRange) {
+      case '1day':
+        range = 1;
+        break;
+      case '1week':
+        range = 7;
+        break;
+      case '1month':
+        range = 30;
+        break;
+      case '1year':
+        setFilteredData([...allData].reverse()); // Reverse the order for latest data on the right
+        return;
+    }
+
+    const newFilteredData = allData
+      .filter((d) => new Date(d.label!) >= new Date(now.getTime() - range * 24 * 60 * 60 * 1000))
+      .reverse(); // Reverse the filtered data
+    setFilteredData(newFilteredData);
+  }, [selectedRange, allData]);
+
+  // Reset to last data point
+  useEffect(() => {
+    if (filteredData.length > 0) {
+      const lastData = filteredData[filteredData.length - 1];
+      setCurrentValue(lastData.value);
+      setCurrentDate(new Date(lastData.label!).toLocaleDateString());
+    }
+  }, [filteredData]);
+
+  const handlePointer = ({ pointerIndex }: { pointerIndex: number }) => {
+    if (pointerIndex === -1) {
+      // Reset to the last data point when pointer is released
+      const lastData = filteredData[filteredData.length - 1];
+      setCurrentValue(lastData?.value || 0);
+      setCurrentDate(new Date(lastData?.label || '').toLocaleDateString());
+    } else if (pointerIndex >= 0 && pointerIndex < filteredData.length) {
+      // Update current value and date when the pointer is active
+      setCurrentValue(filteredData[pointerIndex]?.value || 0);
+      setCurrentDate(new Date(filteredData[pointerIndex]?.label || '').toLocaleDateString());
+    }
+  };
+
+  return (
+    <View
+      className="rounded-xl bg-content p-4 shadow-md"
+      style={{ overflow: 'hidden' }}
+      onLayout={(event) => {
+        const { width } = event.nativeEvent.layout;
+        setContainerWidth(width - 16); // Account for padding inside the frame
+      }}>
+      <FText className="mb-2 !text-2xl" bold>
+        Price History
+      </FText>
```

Comment:
Do you think that we really need this title?

Reply:
We don't need it if it is clear that the graph is for a certain asset or wallet. Maybe in the future it could be an optional param. I will remove for now if you think it's unnecessary.